packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -200,6 +207,42 @@ export class RemoteMessageProcessor {
 		}
 		assert(batchMetadataFlag !== true, 0x9d6 /* Unexpected batch start marker */);
 
+		// Validate that the sequence numbers are contiguous. If there are non-runtime messages in the batch,
+		// the remote message processor isn't called and so it will appear like the batch is missing messages
+		// and this case will be hit.
+		const previousMessage =
+			this.batchInProgress.messages[this.batchInProgress.messages.length - 1];
+		if (
+			previousMessage !== undefined &&
```

Comment:
Looks to me like you can (and should) assert that previousMessage is not undefined